src/data/list/basic.lean

```diff
@@ -856,6 +856,13 @@ theorem nth_le_reverse_aux1 : ∀ (l r : list α) (i h1 h2), nth_le (reverse_cor
 | (a :: l) r i := by rw (show i + length (a :: l) = i + 1 + length l, from add_right_comm i (length l) 1); exact
   λh1 h2, nth_le_reverse_aux1 l (a :: r) (i+1) h1 (succ_lt_succ h2)
 
+lemma index_of_inj [decidable_eq α] {l : list α} {x y : α}
```

Comment:
Would `eq_of_eq_index` be a better name? It's not really using the `injective` predicate...

Reply:
Actually there are several examples of using `f_inj` as a suffix to mean injective, especially when written out as an iff, i.e. `f x = f y <-> x = y`. By contrast, `f_injective` will be used to mean `injective f` using the token.

Reply:
Ok, then I guess this is ready to merge.